sql/catalyst/src/main/scala/org/apache/spark/sql/types/Metadata.scala

```diff
@@ -85,17 +85,18 @@ sealed class Metadata private[types] (private[types] val map: Map[String, Any])
   override def equals(obj: Any): Boolean = {
     obj match {
       case that: Metadata =>
-        if (map.keySet == that.map.keySet) {
-          map.keys.forall { k =>
-            (map(k), that.map(k)) match {
-              case (v0: Array[_], v1: Array[_]) =>
-                v0.view == v1.view
-              case (v0, v1) =>
-                v0 == v1
-            }
+        map.keysIterator.forall { key =>
```

Comment:
Whoops, realized this is wrong if `that.map.keySet` contains keys that aren't in this key set. Will fix now.